sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/TextLineNumber.java

```diff
@@ -0,0 +1,460 @@
+   *                               the minimum width of the component
+   */
+  public TextLineNumber(JTextComponent component, int minimumDisplayDigits) {
+    this.component = component;
+
+    setFont(component.getFont());
+
+    setBorderGap(5);
+    setCurrentLineForeground(Color.RED);
+    setDigitAlignment(RIGHT);
+    setMinimumDisplayDigits(minimumDisplayDigits);
+
+    component.getDocument().addDocumentListener(this);
+    component.addCaretListener(this);
+    component.addPropertyChangeListener("font", this);
+  }
+
+  /**
+   *  Gets the update font property
+   *
+   *  @return the update font property
+   */
+  public boolean getUpdateFont() {
+    return updateFont;
+  }
+
+  /**
+   *  Set the update font property. Indicates whether this Font should be
+   *  updated automatically when the Font of the related text component
+   *  is changed.
+   *
+   *  @param updateFont  when true update the Font and repaint the line
+   *                     numbers, otherwise just repaint the line numbers.
+   */
+  public void setUpdateFont(boolean updateFont) {
+    this.updateFont = updateFont;
+  }
+
+  /**
+   *  Gets the border gap
+   *
+   *  @return the border gap in pixels
+   */
+  public int getBorderGap() {
+    return borderGap;
+  }
+
+  /**
+   *  The border gap is used in calculating the left and right insets of the
+   *  border. Default value is 5.
+   *
+   *  @param borderGap  the gap in pixels
+   */
+  public void setBorderGap(int borderGap) {
+    this.borderGap = borderGap;
+    Border inner = new EmptyBorder(0, borderGap, 0, borderGap);
+    setBorder(new CompoundBorder(OUTER, inner));
+    lastDigits = 0;
+    setPreferredWidth();
+  }
+
+  /**
+   *  Gets the current line rendering Color
+   *
+   *  @return the Color used to render the current line number
+   */
+  public Color getCurrentLineForeground() {
+    return currentLineForeground == null ? getForeground() : currentLineForeground;
+  }
+
+  /**
+   *  The Color used to render the current line digits. Default is Coolor.RED.
+   *
+   *  @param currentLineForeground  the Color used to render the current line
+   */
+  public void setCurrentLineForeground(Color currentLineForeground) {
+    this.currentLineForeground = currentLineForeground;
+  }
+
+  /**
+   *  Gets the digit alignment
+   *
+   *  @return the alignment of the painted digits
+   */
+  public float getDigitAlignment() {
+    return digitAlignment;
+  }
+
+  /**
+   *  Specify the horizontal alignment of the digits within the component.
+   *  Common values would be:
+   *  <ul>
+   *  <li>TextLineNumber.LEFT
+   *  <li>TextLineNumber.CENTER
+   *  <li>TextLineNumber.RIGHT (default)
+   *	</ul>
+   *  @param currentLineForeground  the Color used to render the current line
+   */
+  public void setDigitAlignment(float digitAlignment) {
+    this.digitAlignment = digitAlignment > 1.0f ? 1.0f : digitAlignment < 0.0f ? -1.0f : digitAlignment;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)